### PR TITLE
refactor PMacc plugin slicing

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- rename `TimeSlice` into `Slice`
- add method `toRangeSlice` with special handling for the case that only the begin of the range is given

This PR fixes a bug @PrometheusPi found that setting only the begin of a range for the OpenPMD plugin dumped all data, instead of a single slice.

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
@PrometheusPi can be merged

File: include/pmacc/pluginSystem/toSlice.hpp [hunk 1]
```diff
@@ -53,14 +53,17 @@ namespace pmacc
 
         /** create a TimeSlice out of an string
          *
-         * Parse a comma separated list of time slices and creates a vector of TimeSlices.
-         * TimeSlice Syntax:
+         * Parse a comma separated list of time slices and creates a vector of slices.
+         * time slice syntax:
          *   - `start:stop:period`
          *   - a number ``N is equal to `::N`
```
Comment (PrometheusPi) [inline include/pmacc/pluginSystem/toSlice.hpp:59]: Is this still true? I thought a single Number now maps to `N:N+1:1`.

Comment (PrometheusPi) [reply include/pmacc/pluginSystem/toSlice.hpp:59]: yes - this is the (old) time slice - thus correctly maps to `0:-1:N`

**Review by PrometheusPi — APPROVED** [review]
I am a bit confused with `Slice`, but it looks good to me.

**Pull request merged**